@xen-orchestra/lite/src/components/form/FormNumber.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <FormInput v-model="localValue" inputmode="decimal" />
+</template>
+
+<script lang="ts" setup>
+import FormInput from "@/components/form/FormInput.vue";
+import { computed, ref, watch } from "vue";
+
+const props = defineProps<{
+  modelValue: number | undefined;
+  maxDecimals?: number;
+}>();
+
+const emit = defineEmits<{
+  (event: "update:modelValue", value: number | undefined): void;
+}>();
+
+const localValue = ref("");
+
+const hasTrailingDot = ref(false);
+
+const cleaningRegex = computed(() => {
+  if (props.maxDecimals === undefined) {
+    // Any number with optional decimal part
+    return /(\d*\.?\d*)/;
+  }
+
+  if (props.maxDecimals > 0) {
+    // Numbers with up to `props.maxDecimals` decimal places
+    return new RegExp(`(\\d*\\.?\\d{0,${props.maxDecimals}})`);
+  }
+
+  // Integer numbers only
+  return /(\d*)/;
+});
+
+watch(
+  localValue,
+  (newLocalValue) => {
+    const cleanValue =
+      localValue.value
+        .replace(",", ".")
+        .replace(/[^0-9.]/g, "")
```

Comment:
It is intentional to exclude the `e` character, which can be used in native number input?